.github/workflows/infrautils-cfn-deploy.yml

```diff
@@ -0,0 +1,70 @@
+# This workflow installs infrautils (DevOps maintained infrastructure as service python package) from JFrog
+# and uses infrautils (boto3) to deploy a cloudformation stack update
+# https://github.com/energyhub/infra-utils
+name: Infrautils CFN Deploy
+
+on:
+  workflow_call:
+    secrets:
+      JFROG_CI_PASSWORD:
+        required: true
+    inputs:
+      service:
+        description: "Name of the service to be deployed"
+        required: true
+        type: string
+      image-tag:
+        description: "Tag of the image to deploy"
+        required: true
+        type: string
+      vpc:
+        description: "VPC to deploy to, e.g. 'prod', 'qa-1', 'mec-rc', etc."
+        required: true
+        type: string
+      templatedir:
+        description: "location of the cloudformation template"
+        type: string
+        required: false
+      templateparams:
+        description: "location of the cloudformation parameter overrides file"
+        type: string
+        required: false
+
+env:
+  TEMPLATE_DIR: {{ inputs.templatedir || cloudformation/templates }}
+  TEMPLATE_PARAMS: {{ inputs.templateparams || cloudformation/params }}
+  JFROG_PYPI_URI: ehub.jfrog.io/artifactory/api/pypi/pypi-all/simple
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  infrautils-cfn-deploy:
+    environment: ${{ inputs.environment }}
+    runs-on: ubuntu-latest
+    name: infrautils-cfn-deploy
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+
+      - name: Install infrautils
+        run: |
+          pip install infrautils \
+            -i https://ci-infra-utils:${{ secrets.JFROG_PASSWORD }}@${{ env.JFROG_PYPI_URI }}
+
+      - name: Assume Actions IAM Role
+        uses: energyhub/workflow-actions/assume-actions-role@v2.4.2
+
+      - name: Deploy via Cloudformation
+        run: |
+          infrautils cfn.deploy \
```

Comment:
just looking at the difference between the old deploy script and this...do we need `stack-name` here? Or does that have to be in the cfn template?

Reply:
the infrautils cfn.deploy task can either take stack-name as a cli input argument (-s 'qa-N-super-awesome-service'), or it will look for it in the params yaml file, the key 'StackName', OR if none of those are defined, it will make a name for the stack based on the template filename. The latter isn't desirable because most templates have a generic template filename, and the stack-name should include the vpc it is deployed in. 

the way this is written, it does assume there is a parameter file for parameter overrides. if that's not desirable, I can instead use '-s ${{ inputs.vpc }}-${{ inputs.service}} -e Env=${{ inputs.vpc }}' and figure out another way to stuff the other extraneous parameter overrides in there too.

Reply:
definitely agree. I'm trying to solve the best away around that. Since blue/green is gone we can put all the vpc and api gateway inputs back into the template. we could also bake into templates default number of tasks, cpu/mem for qa vs prod as a conditional to remove the need to add those as parameter overrides as well.

Reply:
@lainekendall i've made some updates - we've got https://github.com/energyhub/java-microservice-template/pull/76 to move VPC/API-Gateway imports back into the template now that we can because blue/green is no longer needed. and I've added an additional input "additional-params" for any extra arguments to be added - it's very clunky because gha inputs are limited to string or bool, so we can't provide a list of key-value pairs and interpolate that into cli inputs for infrautils. so one would set additional-params to exactly the syntax infrautils expects (ie "-e Env=qa-2 -e AppRef=ctp"). I don't love it, but it should work for now to get rid of the deploy shell script. I've taken out the parameter flat file input but can add it back in conditionally (find a way to add the argument to the infrautils deploy command only if the file exists....) but we could also just put anything additional needed in the additional-params input and bake as much into the template as possible.